src/main/java/com/aws/greengrass/clientdevices/auth/certificate/infra/BackgroundCertificateRefresh.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.certificate.infra;
+
+import com.aws.greengrass.clientdevices.auth.api.UseCases;
+import com.aws.greengrass.clientdevices.auth.infra.NetworkState;
+import com.aws.greengrass.clientdevices.auth.iot.CertificateRegistry;
+import com.aws.greengrass.clientdevices.auth.iot.usecases.VerifyIotCertificate;
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.inject.Inject;
+
+/**
+ * Periodically calls the VerifyIotCertificate to update the
+ * locally cached certificates on a cadence.
+ */
+public class BackgroundCertificateRefresh implements Runnable {
+    private final CertificateRegistry registry;
+    private final UseCases useCases;
+    private final NetworkState networkState;
+    private static final int DEFAULT_INTERVAL_SECONDS = 60 * 60; // 1H
+    private static final Logger logger = LogManager.getLogger(BackgroundCertificateRefresh.class);
+
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private final AtomicReference<ScheduledFuture<?>> scheduledFuture = new AtomicReference<>(null);
```

Comment:
I am using an atomic ref here to avoid the issue of checkstyle complaining I can't set values to null... I could add suppressions though if that is the route we want to take.

Reply:
That's not checkstyle complaining... it's just a compiler error, right? If you update this variable then it isn't final.

Reply:
Nope if I don't make it final and set variables to null checkstyle complains